Return 404s for invalid boundary layer requests.

caseypt opened the pull request:
* Previously a 500 error would be returned if an table name wasn't
provided or was invalid. A 404 error is more appropriate for these
cases.

I tested the site out on our Nexus table and my iPhone, and the boundary selection worked fine in both cases. I wasn't able to reproduce this error using the app. I had to visit the bad URL directly.

**Testing instructions**
- Visit http://localhost:8000/api/modeling/boundary-layers/
- A DRF 404 error page should be returned.
- Visiting the same URL or `develop`, production or staging should throw a 500.

Connects to  #899

mmcfarland commented on src/mmw/apps/modeling/views.py:
```diff
@@ -276,13 +276,16 @@ def _construct_tr55_job_chain(model_input, job_id):
+        query = 'SELECT {field} FROM {table} WHERE id = %s'.format(
+                field=json_field, table=table_name)
+    except KeyError:
+        return Response(status=status.HTTP_404_NOT_FOUND)
```
I think it might be useful to distinguish between the two failure scenarios: a request to a non-existent resource and not actually specifying a resource.  Consider checking if `table_code` is not provided and returning early with a `400` in that case.

caseypt replied:
Will do.

caseypt commented:
Updated.

I modified the URL capture logic for this endpoint so that you can't not provide parameters. This takes care of the case of a missing table code or feature id.

I also added a test case for the bad table code scenario.

mmcfarland commented:
+1, tested.
